src/main/java/com/epam/preprod/sirenko/containers/FactoryContainer.java

```diff
@@ -1,22 +1,28 @@
 package com.epam.preprod.sirenko.containers;
 
-import com.epam.preprod.sirenko.*;
+import com.epam.preprod.sirenko.strategy.Strategy;
+import com.epam.preprod.sirenko.strategy.impl.ClothingFactoryImpl;
+import com.epam.preprod.sirenko.strategy.CreateProductFactory;
+import com.epam.preprod.sirenko.strategy.impl.DryFoodFactoryImpl;
+import com.epam.preprod.sirenko.strategy.impl.FoodFactoryImpl;
 
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
 
 public class FactoryContainer {
 	private Map<String, CreateProductFactory> factories = new HashMap<>();
+	private Strategy strategy;
```

Comment:
We dont need strategy as a field here.You can pass it do directly in factories in init method

Reply:
I get strategy from user in Demo and need to pass strategy to FactoryContainer like this: 
FactoryContainer factoryContainer = new FactoryContainer(strategy);
Couldn't find other way to pass strategy other than in field